### PR TITLE
make modules: update, sync, update to fix #136

### LINKED ISSUE #136: git submodule logic changes remotes
I was wondering why the URLs of remotes of submodules where being changed until I realized the issue was with `make modules`.

To reproduce:

```sh
$ cd core
$ git remote add kba https://github.com/kba/ocrd-core
$ git checkout -t kba/logging-test
$ git remote -v
kba     https://github.com/kba/ocrd-core (fetch)
kba     https://github.com/kba/ocrd-core (push)
origin  https://github.com/OCR-D/core.git (fetch)
origin  https://github.com/OCR-D/core.git (push)
```

Now, if I run `make modules` and change back into `core`:

```sh
$ make modules OCRD_MODULES=core
sem --fg --id ocrd_all_git git submodule sync  core
Synchronizing submodule url for 'core'
if git submodule status  core | grep -qv '^ '; then \
        sem --fg --id ocrd_all_git git submodule update --init  core && \
        touch core; fi
Submodule path 'core': checked out '320a2fdeda6836bc4da522620bf169c22d471cbd'
$ cd core
$ git remote -v
kba     https://github.com/OCR-D/core.git (fetch)
kba     https://github.com/OCR-D/core.git (push)
origin  https://github.com/OCR-D/core.git (fetch)
origin  https://github.com/OCR-D/core.git (push)
```

I understand that the `submodule update` will change to the recorded submodule commit.

I **do not** understand how and why the remote URL is changed.

### PR DESCRIPTION
**Pull request by kba**

### CODE REVIEW

**Comment by stweil** [comment]
I rebased the branch to fix a merge conflict.

**Comment by stweil** [comment]
@kba, is it okay if I squash and merge this PR?

**Comment by kba** [comment]
> @kba, is it okay if I squash and merge this PR?

No problem with squashing but let's be sure that this has no unintended side effects before merging. Since the issue it fixes only occurs for me AFAICS, there's no hurry. Getting core and processor updates merged is more important.

**Comment by kba** [comment]
@bertsky are there any as yet unseen consequences to this? If not, then merging soon would really  help me, I've lost most remote URLs in the recent update (#181)...

**Comment by bertsky** [comment]
> are there any as yet unseen consequences to this?

As I said in #136, I don't know really. It will become a little slower obviously, but I don't know what's "right" for our different scenarios (submodules out of date / submodules at custom state / submodules at different remote/branch, versus upstream changes URL / upstream changes commit) and combinations thereof...

But at least we should try to remove the semaphore from `submodule update` – see [here](https://github.com/OCR-D/ocrd_all/issues/169#issuecomment-677915363).

File: Makefile [hunk 1]
```diff
@@ -124,7 +124,7 @@ $(OCRD_MODULES): always-update
 	if git submodule status $(GIT_RECURSIVE) $@ | grep -qv '^ '; then \
 		$(SEM) git submodule update --init $(GIT_RECURSIVE) $@ && \
 		touch $@; fi
-	$(SEM) git submodule sync $(GIT_RECURSIVE) $@
+	git submodule sync $(GIT_RECURSIVE) $@
```
Comment (bertsky) [inline Makefile:127]: This removes the semaphore at the wrong place. We do need it at sync, but not at update

Comment (kba) [reply Makefile:127]: oops, typo. I'll force-push a fix.

Comment (kba) [reply Makefile]: https://github.com/OCR-D/ocrd_all/pull/148/commits/e0fcf2991be6fda86efbf13bc7afc4778ce42fc9

**Comment by kba** [comment]
This is a problem only I encountered AFAICT and I have my workarounds, so no need to change anything that might break stuff/make stuff slower for everybody else.

**Pull request closed without merging**